itm/itm_test.go

```diff
@@ -159,3 +182,37 @@ func TestUserAgentStringOnRequest(t *testing.T) {
 		}
 	}
 }
+
+func compareMaps(expectedMap, receivedMap map[string]interface{}) bool {
+	for key, expectedValue := range expectedMap {
+
+		receivedValue, ok := receivedMap[key]
+		if !ok {
+			return false // Key not found in receivedMap
+		}
+
+		switch expectedValue.(type) {
+		case map[string]interface{}:
+			// Recursively compare nested maps
+			expectedValueNested, ok := expectedValue.(map[string]interface{})
+			if !ok {
+				return false // issue
+			}
+			receivedValueNested, ok := receivedValue.(map[string]interface{})
+			if !ok {
+				return false // Type mismatch
+			}
+			if !compareMaps(expectedValueNested, receivedValueNested) {
+				return false // Nested maps not equal
+			}
+		default:
+			// Compare non-map values
+			if expectedValue != receivedValue {
+				return false // Values not equal
+
+			}
+		}
+	}
+
+	return true
+}
```

Comment:
how about using a library for this https://github.com/google/go-cmp

```
if !cmp.Equal(receivedValue, expectedValue) {
    t.Errorf("Wrong object received, got=%s", cmp.Diff(expectedValue, receivedValue))
}
```

Reply:
In the caller,
if !compareMaps(expectedRequestData, parsedBody) {
			t.Error(unexpectedValueString("Request body", expectedRequestData, parsedBody))
		}

the parsedBody has more entries than the expectedRequestData. 

the compareMaps returns true only if all the entries in expectedRequestData are at least in parsedBody.

the cmp.Equal expects that every entries are in both maps.